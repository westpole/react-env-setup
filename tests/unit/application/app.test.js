import toJson from 'enzyme-to-json';

import { mockComponent } from '../../tools/mock-component.jsx';
import App from '../../../src/scripts/application/app.jsx';

describe('Application/app', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mockComponent({
      type: 'shallow',
      scope: App,
      props: {
        testProp: 'test',
      },
    });
  });

  it('should match saved snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should return a React object for Component as a function', () => {
    const AppMock = App({ testProp: 1 });

    expect(AppMock.props.children.props.testProp).toEqual(1);
    expect(AppMock.props.store).toBeDefined();
  });

  it('should have store object set', () => {
    expect(wrapper.find('[store]').length).toBeDefined();
  });

  it('should set properties for the Application', () => {
    expect(wrapper.find('[testProp="test"]').length).toEqual(1);
  });
});
