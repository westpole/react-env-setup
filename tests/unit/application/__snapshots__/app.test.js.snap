// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application/app should match saved snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(VehiclesComponent)
    testProp="test"
  />
</Provider>
`;
